### PR TITLE
make it more configurable to wait for external view to converge with ideal states

### PR DESCRIPTION
**Pull request by klsince**
added two configs to make it more configurable on how often to check if and how long to wait if external view converges with ideal states

## Release notes ##
added two configs for segmentRelocator to fine tune the periodic table rebalance task
controller.segmentRelocator.externalViewStabilizationTimeoutMs - 1hr by default
controller.segmentRelocator.externalViewCheckIntervalMs - 1s by default

externalViewStabilizationTimeoutMs and externalViewCheckIntervalMs can also be set when triggering rebalance via rest API or cmd line

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java [hunk 1]
```diff
@@ -621,27 +621,31 @@ private ObjectNode validateConfig(TableConfig tableConfig, Schema schema, @Nulla
...
+      boolean downtime, @ApiParam(value =
+      "For no-downtime rebalance, minimum number of replicas to keep alive during rebalance, or maximum "
           + "number of replicas allowed to be unavailable if value is negative") @DefaultValue("1")
-  @QueryParam("minAvailableReplicas") int minAvailableReplicas, @ApiParam(
-      value = "Whether to use best-efforts to rebalance (not fail the rebalance when the no-downtime contract cannot "
-          + "be achieved)") @DefaultValue("false") @QueryParam("bestEfforts") boolean bestEfforts) {
+  @QueryParam("minAvailableReplicas") int minAvailableReplicas, @ApiParam(value =
+      "Whether to use best-efforts to rebalance (not fail the rebalance when the no-downtime contract cannot "
+          + "be achieved)") @DefaultValue("false") @QueryParam("bestEfforts") boolean bestEfforts,
+      @ApiParam(value = "How often to check if external view converges with ideal states") @DefaultValue("1000")
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java:645]: Can you double check the format change? I feel it is not very readable

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java [hunk 2]
```diff
@@ -912,6 +916,14 @@ public boolean enableSegmentRelocatorLocalTierMigration() {
     return getProperty(ControllerPeriodicTasksConf.SEGMENT_RELOCATOR_ENABLE_LOCAL_TIER_MIGRATION, false);
   }
 
+  public long getSegmentRelocatorExternalViewCheckIntervalMs() {
+    return getProperty(ControllerPeriodicTasksConf.SEGMENT_RELOCATOR_EXTERNAL_VIEW_CHECK_INTERVAL_MS, 1000);
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java:920]: Should we use the constant introduced in `RebalanceConfigConstants` as default?

File: pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/RebalanceTableCommand.java [hunk 3]
```diff
@@ -77,6 +77,14 @@ public class RebalanceTableCommand extends AbstractBaseAdminCommand implements C
           + " cannot be achieved, false by default)")
   private boolean _bestEfforts = false;
 
+  @CommandLine.Option(names = {"-externalViewCheckIntervalMs"},
+      description = "How often to check if external view converges with ideal view")
+  private long _externalViewCheckIntervalMs = 1000;
```
Comment (Jackie-Jiang) [inline pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/RebalanceTableCommand.java:82]: Should we use the constant introduced in `RebalanceConfigConstants` as default?

**Review by Jackie-Jiang — APPROVED** [review]
LGTM

**Review by npawar — COMMENTED** [review]
it can get tricky to convey to operator/user, that this timeout should be less than the frequency. could we just reuse frequency? say we use timeout of frequency - 5m ?

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java (hunk 2 above)
Comment (klsince) [reply pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java:920]: yeah, can do. I usually just avoid deps for simple stuff like this.

**Pull request merged**